envoy/config/subscription.h

```diff
@@ -62,9 +62,9 @@ class DecodedResource {
   virtual bool hasResource() const PURE;
 
   /**
-   * @return optional ref<envoy::config::core::v3::Metadata> of a resource.
+   * @return optional envoy::config::core::v3::Metadata of a resource.
    */
-  virtual const OptRef<const envoy::config::core::v3::Metadata> metadata() const PURE;
+  virtual const absl::optional<envoy::config::core::v3::Metadata> metadata() const PURE;
```

Comment:
I am not understanding why this change is needed. This forces a copy at the interface. I think it's OK to have the caller of this observer decide whether to make a copy of it.

Reply:
+1
Why should this be any different than `resource()`, for example?
The caller should ensure that the lifetime of the object is maintained properly.

Reply:
Right, done.